marconi/test/Spec/Utxo.hs

```diff
@@ -22,73 +26,180 @@ import Cardano.Api qualified as C
 import Gen.Cardano.Api.Typed qualified as CGen
 import Marconi.Index.Utxo qualified as Utxo
 import Marconi.Types (CurrentEra, TargetAddresses)
-import RewindableIndex.Index.VSplit qualified as Ix
+import RewindableIndex.Storable (StorableEvent, StorableQuery)
+import RewindableIndex.Storable qualified as Storable
+
+genSlotNo :: Hedgehog.MonadGen m => m C.SlotNo
+genSlotNo = C.SlotNo <$> Gen.word64 (Range.linear 10 1000)
+
+genBlockNo :: Hedgehog.MonadGen m => m C.BlockNo
+genBlockNo = C.BlockNo <$> Gen.word64 (Range.linear 100 1000)
+
+genBlockHeader
+  :: Hedgehog.MonadGen m
+  => m C.BlockNo
+  -> m C.SlotNo
+  -> m C.BlockHeader
+genBlockHeader genB genS = do
+  let validByteSizeLength = 32
+  bs <- Gen.bytes(Range.singleton validByteSizeLength)
+  sn <- genS
+  bn <- genB
+  let (hsh :: C.Hash C.BlockHeader) =  fromJust $ C.deserialiseFromRawBytes(C.proxyToAsType Proxy) bs
+  pure (C.BlockHeader sn hsh bn)
+
+genChainPoint
+  :: Hedgehog.MonadGen m
+  => m C.BlockNo
+  -> m C.SlotNo
+  -> m C.ChainPoint
+genChainPoint genB genS = do
+  (C.BlockHeader sn hsh _) <- genBlockHeader genB genS
+  pure $ C.ChainPoint sn hsh
+
+genChainPoint' :: Hedgehog.MonadGen m => m C.ChainPoint
+genChainPoint' = genChainPoint genBlockNo genSlotNo
+
+genEvents :: Gen (Utxo.StorableEvent Utxo.UtxoHandle)
+genEvents = do
+    txs <- Gen.list (Range.linear 2 5)(CGen.genTx C.ShelleyEra)
+    blkNo  <- genBlockNo
+    cp <- genChainPoint'
+    pure . fromJust $ Utxo.getUtxoEvents Nothing txs blkNo cp
+
+
+headOption :: [a] -> Maybe a
+headOption []    = Nothing
+headOption (x:_) = Just x
 
 tests :: TestTree
-tests = testGroup "Marconi.Index.Specs" $
-    [testPropertyNamed
-     "marconi-index utxos event builder" "Spec. Utxo from Cardano.Api.Tx with targetAddresses"
-     txToUtxoTest
-    , testPropertyNamed
-      "marconi-index utxos event at address" "Spec. Filter UtxoEvents for an address"
-      addressFilteredEventsTest
-    , testPropertyNamed "marconi-index utxos in-memory store" "Spec. Save and retreive UtxoEvents"
+tests = testGroup "Marconi.Utxo.Indexer.Specs are:"
+    [
+     testPropertyNamed "marconi-utxo split-by-address property"
```

Comment:
Would be very helpful to add a comment on each property test function describing what you are testing.